base/cvd/cuttlefish/common/libs/utils/files.cpp

```diff
@@ -129,8 +129,11 @@ Result<void> EnsureDirectoryExists(const std::string& directory_path,
                                                      << strerror(errno));
   }
 
+  CF_EXPECTF(chmod(directory_path.c_str(), mode) == 0,
+            "Failed to set permission on {}: {}", directory_path, strerror(errno));
+
   if (group_name != "") {
-    ChangeGroup(directory_path, group_name);
+    CF_EXPECT(ChangeGroup(directory_path, group_name));
```

Comment:
Please add these changes in aosp too.

Reply:
I'll do that. Thanks for the mentioning :)